xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
We still don't talk about the same thing. You are right that you can't assume the thread to still be running after `IsRunning()` returned true. But sometimes you are interested in the state at exact that one moment. This is for example the case in `CTimer::GetElapsedMilliseconds()` I believe (maybe I'm wrong, I haven't looked at it in detail).